rest/src/main/java/org/jboss/pnc/rest/notifications/model/NotificationFactory.java

```diff
@@ -0,0 +1,34 @@
+package org.jboss.pnc.rest.notifications.model;
+
+import org.jboss.pnc.spi.BuildStatus;
+import org.jboss.pnc.spi.events.BuildStatusChangedEvent;
+
+import javax.enterprise.context.ApplicationScoped;
+import java.util.HashMap;
+import java.util.Map;
+
+@ApplicationScoped
+public class NotificationFactory {
+
+    private Map<BuildStatus, NotificationEventType> externalEvents = new HashMap<>();
+
+    public NotificationFactory() {
+        externalEvents.put(BuildStatus.REPO_SETTING_UP, NotificationEventType.BUILD_STARTED);
```

Comment:
+10000000 Joshua Bloch would be proud sir. Yes I didn't think of that, we don't want implementation details to leak through our API. We should use an abstract set of events for the API that will form part of the API contract.